### PR TITLE
Enhance qx.bom.request.Xhr #9017 : Add support for native reponse and…

### PR DESCRIPTION
**Pull request by hkollmann**
Here is my first try....

### CODE REVIEW

File: framework/source/class/qx/bom/request/Xhr.js [hunk 1]
```diff
@@ -922,15 +934,19 @@ qx.Bootstrap.define("qx.bom.request.Xhr",
...
       if (this.readyState >= qx.bom.request.Xhr.HEADERS_RECEIVED) {
         // In some browsers, XHR properties are not readable
         // while request is in progress.
         try {
           this.status = nxhr.status;
           this.statusText = nxhr.statusText;
-          this.responseText = nxhr.responseText;
-          this.responseXML = nxhr.responseXML;
+          this.response = nxhr.response;
+          if (typeof nxhr.response === "string") {
```
Comment (oetiker) [inline framework/source/class/qx/bom/request/Xhr.js]: are responseXML and responseText not defined if typeof response is not string ?
is this the case for all browsers?

**Comment by oetiker** [comment]
what would be really cool, would be if you could add another example to the top of the file demonstrating how to use the new functionality.

**Comment by oetiker** [comment]
as you can see from the travis-report above, the tests are not happy with your change ... click on 'details' to see where the problem ist ... also, you may want to add additional tests to https://github.com/qooxdoo/qooxdoo/blob/master/framework/source/class/qx/test/bom/request/Xhr.js to check the new functionality you just added.

**Comment by oetiker** [comment]
oh and a note on pull requests ... best is to do

```bash
git checkout -b xhr-pull
```

before committing and pushing ... in that way your new code will be in a separate branch and your pull request will not be from 'master'

when you have fixes for this one. Just push again, the PR in github will automatically update with your next push ... you don't have to modify the PR. But since you are working in your 'master' branch you should make sure to only push changes to the PR and nothing else (hence the hint with the checkout and a separate branch for future PRs, as you may be working on multiple separate PRs).

File: framework/source/class/qx/bom/request/Xhr.js (hunk 1 above)
Comment (hkollmann) [inline framework/source/class/qx/bom/request/Xhr.js]: Yes, if responseTyp is not empty or not "text" an object will be returned.
https://xhr.spec.whatwg.org/#the-responsetype-attribute

**Comment by hkollmann** [comment]
Thanks for the hints, it's my first day with git. So forgive me....

**Comment by hkollmann** [comment]
How can i run the tests locally? I am using an Windows enviroment...

**Comment by hkollmann** [comment]
Is it possible to run the test's with Debugging?

**Comment by hkollmann** [comment]
OK, i looked into the Tests. The poblem is that the tests only work with responseText.
Not with  response. This is ok if you only use responseType="text", In this case the response is copied into the responseText. 
In the test you should set fakeReq.response = "Affe". This could be compared with responseText.

**Comment by oetiker** [comment]
the easiest is just to push an update to the PR, then the travis tests run automatically ... :-) 

you can also use the test runner in your browser http://qooxdoo.github.io/devel/pages/development/frame_apps_testrunner.html

**Comment by oetiker** [comment]
note that your changes should not break existing code, but on the other hand, the tests in qx/test/bom/request/Xhr.js may also bee too simple so your PR should update them too.

**Comment by hkollmann** [comment]
OK, test are running. I added an example in the header.
Writing a new test not possible because FakeXMLHttpRequest only supports string data in his respond method. Sorry for that

**Comment by oetiker** [comment]
I think we should merge this!

**Comment by level420** [comment]
OK merging.
Thank you @hkollmann !

**Comment by level420** [comment]
Merge also approved by @johnspackman

**Pull request merged**